plugins/builds/helper/updateBuild.js

```diff
@@ -241,6 +270,50 @@ function deriveEventStatusFromBuildStatuses(builds) {
     return newEventStatus;
 }
 
+/**
+ * Update stage build's status
+ * Stage build's status can transition as below
+ *   CREATED -> RUNNING -when all builds are succeeded-> SUCCESS
+ *                      -when some builds are terminated-> FAILURE, ABORTED, UNSTABLE, COLLAPSED
+ *
+ * @param {Object} param
+ * @param {StageBuild} param.stageBuild
+ * @param {String} param.newStatus
+ * @param {Job} param.job
+ * @returns {StageBuild} new stage build's status
+ */
+async function updateStageBuildStatus({ stageBuild, newStatus, job }) {
+    if (stageBuild.status === newStatus) {
+        return stageBuild.status;
```

Comment:
I don’t think this causes any issues in the current usage, but it feels a bit inconsistent that the return value of the function can be either a `string` (status) or a `stageBuild` object.

It might be cleaner to always return the `stageBuild` itself and drop the `.status` return to keep the return type consistent.

Reply:
Thank you. I must have overlooked it.